pom.xml

```diff
@@ -101,6 +101,7 @@
     <module>examples</module>
     <module>repl</module>
     <module>launcher</module>
+    <module>external/kafka-0-10-token-provider</module>
```

Comment:
From the JIRA it sounds like a new module is unfortunately the best way to clean this up, so OK. It can't live in the kafka-0-10 module, because non-Kafka things need it? I'm wondering, because of the name.

Reply:
At the moment Structured Streaming uses it but without duplicating the code it must be in a library. This jira/PR is actually a preparation for the DStreams support.

Reply:
All in all kafka-0-10 is DStreams and kafka-0-10-sql is SS and later both need it.

Reply:
Yeah, the new module is specific to kafka. The dependencies from [core](https://github.com/apache/spark/pull/23499/files#diff-357e4854869b2e21c38b1b437f11095aL417) moved into a separate [project](https://github.com/apache/spark/pull/23499/files#diff-b490be28af19c4e6d938d173d4dbec75R50).

[kafka-0-10-sql](https://github.com/apache/spark/pull/23499/files#diff-24cb503f180857e2b7d99c130e61770dR38) is depending on that at the moment. Later kafka-0-10 will also depend on the same thing when the same feature adapted to DStreams.